### PR TITLE
JvmInfoMetrics MeterBinder for jvm.info

### PR DESCRIPTION
**Pull request by ebullient**

### CODE REVIEW

File: micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetrics.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.binder.MeterBinder;
+
+public class JvmInfoMetrics implements MeterBinder {
+
+    @Override
+    public void bindTo(MeterRegistry registry) {
+        Gauge.builder("jvm.info", () -> 1L)
+                .description("JVM version info")
+                .tags("version", System.getProperty("java.runtime.version", "unknown"),
```
[the remark this answers is not in the document]
Comment (ebullient) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetrics.java:28]: The request (via Quarkus) was to surface information they were used to gathering via other means (same attribute).

[the remark this answers is not in the document]
Comment (ebullient) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetrics.java:28]: Also, that is the _version_ not the vendor.

[the remark this answers is not in the document]
Comment (kenfinnigan) [reply micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetrics.java:28]: Apologies, that's what it likely was.

**Review by jonatan-ivanov — COMMENTED** [review]
This is a little bit weird to me since it is not really measuring something.
Usually I put these details into a separate (e.g.: `/info` endpoint).

**Comment by ebullient** [comment]
It was something we were asked for (Quarkus). I'm ambivalent about inclusion, TBH. 
I actually wish we had a separate way to package additional binders (outside of core), so the core library could be a little smaller.

**Comment by hacst** [comment]
The original request for Quarkus was by me. The goal was to get closer to the metrics in the widely used prometheus Java client which produces a jvm_info metric. This metric is quite helpful to correlate jvm and other metrics against the used jvm so questions like whether a new jvm version changes gc/memory/perf behavior of the service can be answered precisely.

**Comment by ebullient** [comment]
Thanks for chipping in! I couldn't remember the whys and wherefores. ;)

**Comment by jonatan-ivanov** [comment]
@hacst Would not it be simpler just attaching these tags to your metrics?
Also, how can you correlate the java version with the heap usage with this?

**Comment by hacst** [comment]
Adding these labels to all metrics of your service is quite cumbersome and has other problems. The idea behind metrics like `jvm_info` is that you can do queries to use them with any other metric later on. E.g. in prometheus
```
somemetriclikememory * on (instance, job) group_left(version) jvm_info
```
Gives you `somemetriclikememory` with the `version` label from `jvm_info`. This is a common pattern with versions (and other data of that kind) in prometheus. See https://www.robustperception.io/exposing-the-software-version-to-prometheus for more explanation.

**Comment by jonatan-ivanov** [comment]
Thanks, that query helped to put the missing part in place. :)

File: micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetrics.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2020 VMware, Inc.
```
Comment (jonatan-ivanov) [inline micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetrics.java:2]: 2021? :)

File: micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetricsTest.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2019 VMware, Inc.
```
Comment (jonatan-ivanov) [inline micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetricsTest.java:2]: 2021? :)

**Review by jonatan-ivanov — CHANGES_REQUESTED** [review]

File: micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetricsTest.java (hunk 3 above)
Comment (ebullient) [reply micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetricsTest.java:2]: I feel like I wrote that .. a long time ago.. LOL

**Review by jonatan-ivanov — APPROVED** [review]

**Pull request merged**